src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -68,6 +69,11 @@ internal sealed class SourcePropertySymbol : PropertySymbol, IAttributeTargetSym
             _syntaxRef = syntax.GetReference();
             _refKind = syntax.Type.GetRefKind();
 
+            if (_refKind == RefKind.RefReadOnly)
+            {
+                bodyBinder.Compilation.EnsureIsReadOnlyAttributeExists(diagnostics, syntax.Type.Location, modifyCompilation: true);
```

Comment:
>bodyBinder.Compilation.EnsureIsReadOnlyAttributeExists(diagnostics, syntax.Type.Location, modifyCompilation: true); [](start = 16, length = 115)

As I mentioned during the first review pass, we should not be doing this eagerly in a constructor. The place where we bind type would be a good place, but we are doing this in two places in this class.